mssmt/store.go

```diff
@@ -244,29 +252,66 @@ func (d *DefaultStore) DeleteCompactedLeaf(key NodeHash) error {
 	return nil
 }
 
+// DeleteRoot deletes the root node of the MS-SMT.
+func (d *DefaultStore) DeleteRoot() error {
+	d.root = nil
+	d.cntDeletes++
+
+	return nil
+}
+
+// DeleteAllNodes deletes all nodes in the MS-SMT.
+func (d *DefaultStore) DeleteAllNodes() error {
+	// Delete leaves, then compacted leaves, then branches.
+	leafCount := len(d.leaves)
+	maps.Clear(d.leaves)
+	d.cntDeletes += leafCount
+
+	compactedLeafCount := len(d.compactedLeaves)
+	maps.Clear(d.compactedLeaves)
+	d.cntDeletes += compactedLeafCount
+
+	branchCount := len(d.branches)
+	maps.Clear(d.branches)
+	d.cntDeletes += branchCount
+
+	return nil
+}
+
 // GetChildren returns the left and right child of the node keyed by the given
 // NodeHash.
 func (d *DefaultStore) GetChildren(height int, key NodeHash) (
 	Node, Node, error) {
 
+	EmptyNodeHash := EmptyTree[height].NodeHash()
 	getNode := func(height uint, key NodeHash) Node {
-		if key == EmptyTree[height].NodeHash() {
+		if key == EmptyNodeHash {
```

Comment:
This changes the behavior of this whole function if I understand correctly. Because we're no longer comparing with the correct empty hash. We call `getNode` with `height` but then further down with `height+1`. So without any additional context, this change looks wrong to me.

Reply:
Ah yes, didn't realize emptyHash would be compared twice - so I think moving it back into getNode() would match the old behavior?